src/test/java/io/pravega/connectors/flink/FlinkPravegaOutputFormatTest.java

```diff
@@ -0,0 +1,191 @@
+        FlinkPravegaOutputFormat.<String>builder()
+                .withEventRouter(eventRouter)
+                .withPravegaConfig(pravegaConfig)
+                .forStream(stream)
+                .build();
+    }
+
+    /**
+     * Testing the builder for right configurations.
+     * Should fail since we don't pass {@link PravegaEventRouter}
+     */
+    @Test(expected = NullPointerException.class)
+    public void testBuilderForFailure2() {
+        FlinkPravegaOutputFormat.<String>builder()
+                .withSerializationSchema(serializationSchema)
+                .withPravegaConfig(pravegaConfig)
+                .forStream(stream)
+                .build();
+    }
+
+    /**
+     * Testing the builder for right configurations.
+     * Should fail since we don't pass {@link Stream}
+     */
+    @Test(expected = IllegalStateException.class)
+    public void testBuilderForFailure3() {
+        FlinkPravegaOutputFormat.<String>builder()
+                .withEventRouter(eventRouter)
+                .withSerializationSchema(serializationSchema)
+                .withPravegaConfig(pravegaConfig)
+                .build();
+    }
+
+    /**
+     * Test case to validate the lifecycle methods of {@link OutputFormat}
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testLifecycleMethods() throws Exception  {
+        ClientConfig clientConfig = ClientConfig.builder().build();
+        when(pravegaConfig.getClientConfig()).thenReturn(clientConfig);
+        when(pravegaConfig.resolve(anyString())).thenReturn(stream);
+
+        EventStreamWriter<String> pravegaWriter = mockEventStreamWriter();
+        ClientFactory clientFactory = mockClientFactory(pravegaWriter);
+        FlinkPravegaOutputFormat<String> spyFlinkPravegaOutputFormat = spyFlinkPravegaOutputFormat(clientFactory);
+
+        // test open
+        spyFlinkPravegaOutputFormat.open(0, 1);
+        verify(spyFlinkPravegaOutputFormat).open(0, 1);
+
+        CompletableFuture<Void> writeFuture = new CompletableFuture<>();
+        when(pravegaWriter.writeEvent(anyString(), anyObject())).thenReturn(writeFuture);
+
+        ExecutorService executorService = spy(new DirectExecutorService());
+        Mockito.doReturn(executorService).when(spyFlinkPravegaOutputFormat).createExecutorService();
+
+        // test writeRecord
+        spyFlinkPravegaOutputFormat.writeRecord("test");
+        verify(spyFlinkPravegaOutputFormat).writeRecord("test");
+        writeFuture.complete(null);
+
+        // test writeRecord for failure
+        doThrow(new IOException()).when(spyFlinkPravegaOutputFormat).checkWriteError();
+        try {
+            spyFlinkPravegaOutputFormat.writeRecord("test");
+        } catch (Exception e) {
+            Assert.assertTrue(e instanceof IOException);
+        }
+
+        // test close error
+        try {
+            spyFlinkPravegaOutputFormat.close();
+            verify(spyFlinkPravegaOutputFormat).close();
```

Comment:
Shouldn't you test that the `close` method has the intended behavior, e.g. that the factory is closed.

Reply:
(done)